### PR TITLE
Convert `ImapResponseParserTest` to Kotlin

### PR DESCRIPTION
**Pull request by cketti**

### CODE REVIEW

File: mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+        assertThat(response).containsExactly("")
+        assertThatAllInputWasConsumed()
+    }
+
+    @Test(expected = IOException::class)
+    fun `readResponse() should throw when end of stream is reached while reading literal`() {
+        val parser = createParserWithData("* {4}\r\nabc")
+
+        parser.readResponse()
+    }
```
Comment (wmontwe) [inline mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt:241]: I think we should stick to one failure testing method and use assertk here too.

```suggestion
    @Test
    fun `readResponse() should throw when end of stream is reached while reading literal`() {
        val parser = createParserWithData("* {4}\r\nabc")

        assertThat {
            parser.readResponse()
        }.isFailure()
            .isInstanceOf(IOException::class)
    }
```

File: mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+
+        val response = parser.readResponse()
+
+        assertThat(response).hasSize(2)
+        assertThat(response).transform { it.getNumber(0) }.isEqualTo(23)
+        assertThat(response).transform { it.getString(1) }.isEqualTo("EXISTS")
+        assertThatAllInputWasConsumed()
+    }
+
+    @Test(expected = IOException::class)
```
Comment (wmontwe) [inline mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt:198]: I think we should stick to one failure testing method and use assertk here too.

File: mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+
+        assertThat {
+            parser.readResponse(callback)
+        }.isFailure()
+            .isEqualTo(exception)
+    }
+
+    @Test
+    fun `readResponse() with quoted string containing an escaped quote character`() {
+        val parser = createParserWithResponses("""* "qu\"oted"""")
```
Comment (wmontwe) [inline mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt:349]: Just noticed that this changed to a raw string, which is better readable, while for example line 424 changed to single quotes with escape characters.

**Review by wmontwe — APPROVED** [review]
Good improvement, expecially in terms of readable test methods.

Just added some nits.

File: mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt (hunk 1 above)
Comment (cketti) [reply mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt]: That was my intention. Thanks for catching the instances I missed :+1:

Fixed in 22e499301db6ba54147a3c55169ab3d9ebe635d8

File: mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt (hunk 3 above)
Comment (cketti) [reply mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt:355]: In 22e499301db6ba54147a3c55169ab3d9ebe635d8 I converted a couple of other string to raw strings.

This instance in particular was a hard call. Using a raw string makes it easier to see the non-Java escaping. However, it also obscures the fact that the last character in the string is a quote character.

Comment (wmontwe) [reply mail/protocols/imap/src/test/java/com/fsck/k9/mail/store/imap/ImapResponseParserTest.kt:355]: Fine with me

**Pull request merged**